addons/io/org.openhab.io.azureiothub/src/main/java/org/openhab/io/internal/azureiothub/CloudClient.java

```diff
@@ -0,0 +1,169 @@
+                logger.error("failed to close IoT client", e);
+            }
+        }
+    }
+
+    private EventHubClient receiveMessages(final String partitionId) {
+        EventHubClient client = null;
+        try {
+            client = EventHubClient.createFromConnectionStringSync(this.connectionstring);
+        } catch (Exception e) {
+            logger.error("Failed to create IoT EventHub client", e);
+            return null;
+        }
+        try {
+            client.createReceiver(EventHubClient.DEFAULT_CONSUMER_GROUP_NAME, partitionId, Instant.now())
+                    .thenAccept(new Consumer<PartitionReceiver>() {
+                        @Override
+                        public void accept(PartitionReceiver receiver) {
+                            logger.debug("Created IOT Hub receiver on partition {}", partitionId);
+                            try {
+                                while (true) {
+                                    Iterable<EventData> receivedEvents = receiver.receive(100).get();
+                                    int batchSize = 0;
+                                    if (receivedEvents != null) {
+                                        for (EventData receivedEvent : receivedEvents) {
+                                            logger.debug("Offset: {}, SeqNo: {}, EnqueueTime: {}",
+                                                    receivedEvent.getSystemProperties().getOffset(),
+                                                    receivedEvent.getSystemProperties().getSequenceNumber(),
+                                                    receivedEvent.getSystemProperties().getEnqueuedTime());
+                                            String itemName = receivedEvent.getSystemProperties()
+                                                    .get("iothub-connection-device-id").toString();
+                                            logger.debug("| device ID: {}", itemName);
+                                            String payload = new String(receivedEvent.getBytes(),
+                                                    Charset.defaultCharset());
+                                            logger.debug("| Message Payload: {}", payload);
+                                            eventPublisher.post(ItemEventFactory.createCommandEvent(itemName,
+                                                    new StringType(payload)));
+                                            batchSize++;
+                                        }
+                                    }
+                                    logger.debug("Partition: {}, ReceivedBatch Size: {}", partitionId, batchSize);
+                                }
+                            } catch (Exception e) {
+                                logger.debug("Failed to receive messages ", e);
+                            }
+                        }
+                    });
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception